typed-racket-lib/typed-racket/utils/mutated-vars.rkt

```diff
@@ -17,8 +19,18 @@
         (loop stx tbl)))
     (syntax-parse stx
       #:literal-sets (kernel-literals)
+      ;; let us care about custom syntax classes
+      [form
```

Comment:
Because we want to know when the field mutator function is called and invalidate occurrence typing in that case.